api/v1/policy_webhook.go

```diff
@@ -29,46 +30,61 @@ var (
 func (r *Policy) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithValidator(&PolicyCustomValidator{}).
 		Complete()
 }
 
-//+kubebuilder:webhook:path=/validate-policy-open-cluster-management-io-v1-policy,mutating=false,failurePolicy=Ignore,sideEffects=None,groups=policy.open-cluster-management.io,resources=policies,verbs=create,versions=v1,name=policy.open-cluster-management.io.webhook,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/validate-policy-open-cluster-management-io-v1-policy,mutating=false,failurePolicy=Ignore,sideEffects=None,groups=policy.open-cluster-management.io,resources=policies,verbs=create,versions=v1,name=policy.open-cluster-management.io.webhook,admissionReviewVersions=v1
+// +kubebuilder:object:generate=false
+type PolicyCustomValidator struct{}
 
-var _ webhook.Validator = &Policy{}
+var _ webhook.CustomValidator = &PolicyCustomValidator{}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *Policy) ValidateCreate() (admission.Warnings, error) {
-	log := policylog.WithValues("policyName", r.Name, "policyNamespace", r.Namespace)
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (r *PolicyCustomValidator) ValidateCreate(_ context.Context, obj runtime.Object) (admission.Warnings, error) {
+	policy, ok := obj.(*Policy)
+	log := policylog.WithValues("policyName", policy.Name, "policyNamespace", policy.Namespace)
```

Comment:
If the type assertion failed, I think it'd result in a `nil` dereference panic?

```suggestion
	log := policylog.WithValues("policyName", obj.Name, "policyNamespace", obj.Namespace)
```

Reply:
oh!! very good catch